apps/common/utils/format.test.ts

```diff
@@ -0,0 +1,157 @@
+		assert.equal(T.assertValidNumber(13, 10, ''), 13, 'Value within the range should be returned as is');
+		assert.equal(T.assertValidNumber(18, 10, ''), 18, 'Value equal to the upper limit (18) should be returned as is');
+		assert.equal(T.assertValidNumber(-1, 7, ''), 7, 'Negative value should fallback to default value');
+		assert.equal(T.assertValidNumber(1.5, 10, ''), 10, 'Floating-point value within the range should fallback');
+	});
+
+	it('is ok for formatLocalAmount', (): void => {
+		assert.equal(
+			T.formatLocalAmount(1234.5678, 2, '$', T.defaultOptions).normalize('NFKC'),
+			'1 234,57 $',
+			'Currency symbol should be displayed'
+		);
+		assert.equal(
+			T.formatLocalAmount(1234.5678, 2, '$', T.defaultOptions).normalize('NFKC'),
+			'1 234,57 $',
+			'Currency symbol should be displayed'
+		);
+		assert.equal(
+			T.formatLocalAmount(1234.5678, 2, 'USD', T.defaultOptions).normalize('NFKC'),
+			'1 234,57 $',
+			'USD currency symbol should be displayed'
+		);
+		assert.equal(
+			T.formatLocalAmount(1234.5678, 2, 'DAI', T.defaultOptions).normalize('NFKC'),
+			'1 234,57 DAI',
+			'DAI currency symbol should be displayed'
+		);
+		assert.equal(
+			T.formatLocalAmount(0.123, 2, 'PERCENT', T.defaultOptions).normalize('NFKC'),
+			'12,30 %',
+			'Percent symbol should be displayed'
+		);
+		assert.equal(
+			T.formatLocalAmount(1234.5678, 2, '', T.defaultOptions).normalize('NFKC'),
+			'1 234,57',
+			'Empty symbol should not affect formatting'
+		);
+		assert.equal(
+			T.formatLocalAmount(1234.5678, 2, '', {shouldDisplaySymbol: false}).normalize('NFKC'),
+			'1 234,568',
+			'Empty symbol should not affect formatting'
+		);
+		assert.equal(
+			T.formatLocalAmount(12345.6789, 2, '$', T.defaultOptions).normalize('NFKC'),
+			'12,35 k $',
+			'Amount should be formatted in short notation'
+		);
+		assert.equal(
+			T.formatLocalAmount(12345678.9, 2, 'USD', T.defaultOptions).normalize('NFKC'),
+			'12,35 M $',
+			'Amount should be formatted in short notation'
+		);
+		assert.equal(
+			T.formatLocalAmount(0.00000123, 8, '$', T.defaultOptions).normalize('NFKC'),
+			'0,00000123 $',
+			'Amount should be formatted with the specified number of decimals'
+		);
+		assert.equal(
+			T.formatLocalAmount(0.00000000123, 12, 'USD', T.defaultOptions).normalize('NFKC'),
+			'0,00000000123 $',
+			'Amount should be formatted with the specified number of decimals'
+		);
+		assert.equal(
+			T.formatLocalAmount(0.01, 2, 'USD', T.defaultOptions).normalize('NFKC'),
+			'0,01 $',
+			'Amount above 0.01 should be formatted as is'
+		);
+		assert.equal(
+			T.formatLocalAmount(0.001, 2, 'OPT', T.defaultOptions).normalize('NFKC'),
+			'0,001 OPT',
+			'Amount above 0.01 should be formatted as is'
+		);
+		assert.equal(
+			T.formatLocalAmount(0.000000000000123, 18, 'YFI', T.defaultOptions).normalize('NFKC'),
+			'0,000000000000123 YFI',
+			'Amount should be formatted with the specified number of decimals'
+		);
+
```

Comment:
We need some edge cases, like when the number is too small to be formatted for example:

```suggestion
		assert.equal(
			T.formatLocalAmount(0.000000000000000000123, 18, 'YFI', T.defaultOptions).normalize('NFKC'),
			'0,00 YFI',
			'Amount should be formatted with the specified number of decimals'
		);
```